data-prepper-script/src/test/java/org/opensearch/dataprepper/script/parser/ScriptParserTest.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.script.parser;
+
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.antlr.v4.runtime.tree.ParseTreeWalker;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+import org.opensearch.dataprepper.script.parser.util.ListenerMatcher;
+import org.opensearch.dataprepper.script.parser.util.TestListener;
```

Comment:
These tests at a high level look good, and I can see that having a `TestListener` makes testing the `ScriptParser` a lot easier. However, it is a little concerning to me that we are heavily relying on two classes here to test the `ScriptParser`, yet we have no isolated testing for the `TestListener` or `ListenerMatcher`, which contain a significant amount of code. Even though `TestListener` is sort of getting tested here when we compare the final results, is there anyway to isolate the testing of `LogListener` somewhere? How do we know for sure that these two classes are performing perfectly for these tests?